src/sources/internal_metrics.rs

```diff
@@ -44,12 +33,15 @@ impl SourceConfig for InternalMetricsConfig {
         shutdown: ShutdownSignal,
         out: Pipeline,
     ) -> crate::Result<super::Source> {
-        Ok(Box::pin(run(
-            get_controller()?,
-            self.scrape_interval_secs,
-            out,
-            shutdown,
-        )))
+        let interval = time::Duration::from_secs(self.scrape_interval_secs);
+        if interval < time::Duration::from_millis(500) {
+            warn!(
+                "Interval set too low ({} secs), use interval >= 0.5 secs.",
+                interval.as_secs_f64()
```

Comment:
```suggestion
```

It seems like this is a no-op since the value is never used?

Reply:
Yep, together with the code suggestion above. You can create multi-line suggestions btw - just press on the line number and drag up or down when creating a comment!

Reply:
Aha! I didn't know that. Thanks!